packages/base-driver/lib/basedriver/commands/bidi.ts

```diff
@@ -0,0 +1,37 @@
+import type {Constraints, Driver, IBidiCommands} from '@appium/types';
+import type {BaseDriver} from '../driver';
+import {mixin} from './mixin';
+
+declare module '../driver' {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  interface BaseDriver<C extends Constraints> extends IBidiCommands {}
+}
+
+const BidiCommands: IBidiCommands = {
+  async bidiSubscribe<C extends Constraints>(
+    this: BaseDriver<C>,
+    events: string[],
+    contexts: string[] = [''],
+  ) {
+    for (const event of events) {
+      this.bidiEventSubs[event] = contexts;
+    }
+  },
+
+  async bidiUnsubscribe<C extends Constraints>(
+    this: BaseDriver<C>,
+    events: string[],
+    contexts: string[] = [''],
+  ) {
+    for (const event of events) {
+      if (this.bidiEventSubs[event]) {
+        this.bidiEventSubs[event] = this.bidiEventSubs[event].filter((c) => !contexts.includes(c));
```

Comment:
not sure how `includes` reacts to an empty string

Reply:
```
> [''].includes('')
true
```